File: app/src/main/java/com/muliulan/mll/dada/dawdawda/MainActivity.java
//package com.muliulan.mll.dada.dawdawda;
//
//import android.os.Bundle;
//import android.util.Log;
//import android.view.View;
//
//import com.muliulan.mll.R;
//
//import androidx.appcompat.app.AppCompatActivity;
//
//public class MainActivity extends AppCompatActivity {
//
//    private static final String TAG = "MainActivity";
//
//    @Override
//    protected void onCreate(Bundle savedInstanceState) {
//        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_main);
//        InjectUtils.injectEvent(this);
//    }
//
//    @OnClick({R.id.btn1, R.id.btn2})
//    public void click(View view) {
//        switch (view.getId()) {
//            case R.id.btn1:
//                Log.i(TAG, "click: 按钮1");
//                break;
//            case R.id.btn2:
//                Log.i(TAG, "click: 按钮2");
//                break;
//        }
//    }
//
//
//    @OnLongClick({R.id.btn1, R.id.btn2})
//    public boolean longClick(View view) {
//        switch (view.getId()) {
//            case R.id.btn1:
//                Log.i(TAG, "longClick: 按钮1");
//                break;
//            case R.id.btn2:
//                Log.i(TAG, "longClick: 按钮2");
//                break;
//        }
//        return false;
//    }
//}

File: app/src/main/java/com/shopping/leetcode/MainActivity.kt
package com.shopping.leetcode

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import com.shopping.leetcode.code.Main22
import com.shopping.leetcode.code.Main23
import com.shopping.leetcode.offer.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val currentTimeMillis = System.currentTimeMillis()

        val codeAA = Offer30().aa()
        Log.e("mll_codeAA", "$codeAA   ${System.currentTimeMillis() - currentTimeMillis}")

//        val offerAA = Offer6().aa()
//        Log.e("mll_offerAA", "$offerAA   ${System.currentTimeMillis() - currentTimeMillis}")
//        Log.e("mll_codeaaaaaa","${System.nanoTime()}      ${  System.currentTimeMillis()}")


//        val fd = contentResolver.openFileDescriptor(uri, "r")
//        if (fd != null) {
//            val bitmap = BitmapFactory.decodeFileDescriptor(fd.fileDescriptor)
//            fd.close()
//            imageView.setImageBitmap(bitmap)
//        }


    }


}